p2p/resolve_ref.go

```diff
@@ -2,88 +2,186 @@ package p2p
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
+	"sync"
+	"time"
 
-	"github.com/qri-io/qri/repo"
-	reporef "github.com/qri-io/qri/repo/ref"
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/qri-io/qri/dsref"
 )
 
-// MtResolveDatasetRef resolves a dataset reference
-const MtResolveDatasetRef = MsgType("resolve_dataset_ref")
+const (
+	// p2pRefResolverTimeout is the length of time we will wait for a
+	// RefResolverRequest response before cancelling the context
+	// this can potentially be a config option in the future
+	p2pRefResolverTimeout = time.Second * 60
```

Comment:
let's drop this to 20 seconds. We're only firing requests to all connected peers, not yet doing any fancy re-firing as new peers connect or anything, so there's very little reason to have this be longer than 20 seconds.

Also, ref resolution is supposed to (hopefully) be hella fast.